### PR TITLE
*(ticdc): make some linter happy

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/7271

### What is changed and how it works?
Bump the golang-lint-ci to 1.50.0 locally and try to fix some lint issues.

#### Tests 

Bump the golang-lint-ci to 1.50.0 and `make check` for ticdc.
##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

File: cdc/api/v1/api.go [hunk 1]
```diff
@@ -173,10 +173,8 @@ func (h *OpenAPI) ListChangefeed(c *gin.Context) {
 			ID:         cfID.ID,
 		}
 
-		if cfInfo != nil {
-			resp.FeedState = cfInfo.State
-			resp.RunningError = cfInfo.Error
-		}
+		resp.FeedState = cfInfo.State
```
Comment (hi-rustin) [inline cdc/api/v1/api.go:176]: /cc @asddongmen 
Please take a look. We directly use it above but check nil in here.

File: cdc/sinkv2/tablesink/table_sink_impl.go [hunk 2]
```diff
@@ -70,11 +72,13 @@ func New[E eventsink.TableEvent](
 	}
 }
 
+// AppendRowChangedEvents implements the TableSink interface.
```
Comment (hi-rustin) [inline cdc/sinkv2/tablesink/table_sink_impl.go:75]: I'll update this.

File: pkg/p2p/server.go [hunk 3]
```diff
@@ -767,20 +755,20 @@ func (m *MessageServer) receive(stream p2p.CDCPeerToPeer_SendMessageServer, stre
 	}
 }
 
-func (m *MessageServer) handleMessage(ctx context.Context, streamMeta *p2p.StreamMeta, entry *p2p.MessageEntry) error {
+func (m *MessageServer) handleMessage(ctx context.Context, streamMeta *p2p.StreamMeta, entry *p2p.MessageEntry) {
```
Comment (hi-rustin) [inline pkg/p2p/server.go:758]: /cc @overvenus 
Could you please take a look? Does this change OK?

File: pkg/upstream/upstream.go [hunk 4]
```diff
@@ -215,7 +215,7 @@ func initUpstream(ctx context.Context, up *Upstream, gcServiceID string) error {
 // Close all resources.
 func (up *Upstream) Close() {
 	up.mu.Lock()
-	up.mu.Unlock()
+	defer up.mu.Unlock()
```
Comment (hi-rustin) [inline pkg/upstream/upstream.go:218]: /cc @asddongmen 
Does this make sense?

**Review by hi-rustin — COMMENTED** [review]
Self check

File: cdc/sinkv2/tablesink/table_sink_impl.go (hunk 2 above)
Comment (hi-rustin) [reply cdc/sinkv2/tablesink/table_sink_impl.go]: Done

**Comment by hi-rustin** [comment]
I am not going to bump the version in this PR. Because the `engine` also needs to fix some linter issues.

File: pkg/upstream/upstream.go (hunk 4 above)
Comment (asddongmen) [reply pkg/upstream/upstream.go:218]: yep, it's a bug.

**Review by asddongmen — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Comment by hi-rustin** [comment]
Thanks for your review!  💚 💙 💜 💛 ❤️

**Pull request merged**